### PR TITLE
feat(#8181): add automation tests for report's filter

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

[We removed the old search and filters](https://github.com/medic/cht-core/pull/9326). These features are well covered with unit tests, but we need automation tests for form, facility, and status filters. This PR adds basic test cases for those three filters.  

- There are already automation for [Reports' search](https://github.com/medic/cht-core/blob/master/tests/e2e/default/reports/search-reports.wdio-spec.js) and [Contacts' search](https://github.com/medic/cht-core/blob/master/tests/e2e/default/contacts/search-contacts.wdio-spec.js)
- There is already automation [for the date filter](https://github.com/medic/cht-core/blob/master/tests/e2e/default/reports/sidebar-filter.wdio-spec.js)

#8181
https://github.com/medic/care-teams/issues/148

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8181-e2e-remove-old-design-filter/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8181-e2e-remove-old-design-filter/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8181-e2e-remove-old-design-filter/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@kapelo @Benmuiruri this is ready for review, the new test cases are passing consistently. With this, all filters in the sidebar have a test case.

**Review by kapelo — COMMENTED** [review]
@latin-panda I ran the test on my local and these 3 tests fail consistently
- should filter by form
- should filter by place
- should filter by status
sidebar test

File: tests/page-objects/default/reports/reports.wdio.page.js [hunk 1]
```diff
@@ -252,6 +258,36 @@ const openSidebarFilterDateAccordion = async () => {
   return (await sidebarFilterDateAccordionBody()).waitForDisplayed();
 };
 
+const filterByForm = async (formName) => {
+  await (await sidebarFilterFormAccordionHeader()).click();
+  await (await sidebarFilterFormAccordionBody()).waitForDisplayed();
+  const option = sidebarFilterFormAccordionBody().$(`a*=${formName}`);
+  await (await option).waitForDisplayed();
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply tests/page-objects/default/reports/reports.wdio.page.js:265]: Done! Thanks for the suggestion!

File: tests/page-objects/default/reports/reports.wdio.page.js [hunk 2]
```diff
@@ -252,6 +258,36 @@ const openSidebarFilterDateAccordion = async () => {
...
+  await (await option).click();
+};
+
+const filterByStatus = async (statusOption) => {
+  await (await sidebarFilterStatusAccordionHeader()).click();
+  await (await sidebarFilterStatusAccordionBody()).waitForDisplayed();
+  const option = sidebarFilterStatusAccordionBody().$(`a*=${statusOption}`);
+  await (await option).waitForDisplayed();
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply tests/page-objects/default/reports/reports.wdio.page.js:273]: Done

**Comment by latin-panda** [comment]
@kapelo Thanks for the suggestions! Please have a look again

When running local: 

Screenshot 2024-08-19 at 8 14 28 PM

**Review by Benmuiruri — APPROVED** [review]
Good stuff ... LGTM 
Screenshot 2024-08-19 at 16 52 39

**Pull request merged**